lang/test/org/partiql/lang/eval/EvaluatingCompilerDateTimeTests.kt

```diff
@@ -34,4 +63,187 @@ class EvaluatingCompilerDateTimeTests : EvaluatorTestBase() {
             case("'2021-03-17' IS DATE", "false")
         )
     }
+
+    private fun createIonTimeStruct(value: Time): IonStruct =
+        ion.newEmptyStruct().apply {
+            add("hour", ion.newInt(value.hour))
+            add("minute", ion.newInt(value.minute))
+            add("second", ion.newFloat( value.second + value.nano * 10.0.pow(-9)))
+            add("timezone_hour", ion.newInt(value.tz_minutes?.div(60)))
+            add("timezone_minute", ion.newInt(value.tz_minutes?.rem(60)))
+            addTypeAnnotation("\$partiql_time")
+        }
+
+    data class TimeTestCase(val query: String, val expected: String, val expectedTime: Time? = null)
+
+    @ParameterizedTest
+    @ArgumentsSource(ArgumentsForTimeLiterals::class)
+    fun testTime(tc: TimeTestCase)  {
+        val originalExprValue = eval(tc.query)
+        assertEquals(originalExprValue.toString(), tc.expected)
+        if (originalExprValue.type == ExprValueType.TIME) {
+            val timeIonValue = originalExprValue.ionValue
+            timeIonValue as IonStruct
+            assertNotNull(tc.expectedTime)
+            assertEquals("Unexpected ionStruct for time", createIonTimeStruct(value = tc.expectedTime!!), timeIonValue)
+        }
+    }
+
+    /**
+     * Tests to visualize the behavior of evaluation of TIME literals. More tests are covered by [timeLiteralsTests].
+     */
+    private class ArgumentsForTimeLiterals : ArgumentsProviderBase() {
+        private val LOCAL_TIMEZONE_OFFSET = ZoneOffset.systemDefault().rules.getOffset(Instant.now())
+        private val LOCAL_TZ_MINUTES = LOCAL_TIMEZONE_OFFSET.totalSeconds / 60
+
+        private fun case(query: String, expected: String, expectedTime: Time? = null) = TimeTestCase(query, expected, expectedTime)
+
+        override fun getParameters() = listOf(
+            case("TIME '00:00:00.000'", "00:00:00.000000000", Time(0, 0, 0)),
+            case("TIME '23:59:59.99999999'", "23:59:59.999999990", Time(23, 59, 59, 999999990)),
+            case("TIME (2) '23:59:59.99999999'", "00:00:00.000000000", Time(0, 0, 0)),
+            case("TIME '00:45:13.840800524'", "00:45:13.840800524", Time(0, 45, 13, 840800524)),
+            case("TIME '05:20:52.015779149'", "05:20:52.015779149", Time(5, 20, 52, 15779149)),
+            case("TIME '23:59:59'", "23:59:59.000000000", Time(23, 59, 59)),
+            case("TIME (12) '12:24:12.123'", "12:24:12.123000000", Time(12, 24, 12, 123000000)),
+            case("TIME (0) '12:59:59.9'", "13:00:00.000000000", Time(13, 0,0)),
+            case("TIME WITH TIME ZONE '00:00:00'", "00:00:00.000000000${LOCAL_TIMEZONE_OFFSET.getOffsetHHmm()}", Time(0,0,0,0,null, LOCAL_TZ_MINUTES)),
```

Comment:
This literally changes precision, and I don't think that is right (see earlier comments).

Reply:
The new wrapper class preserves the originally specified precision.